client/v2/client.go

```diff
@@ -61,6 +70,9 @@ type Client interface {
 	// Query makes an InfluxDB Query on the database. This will fail if using
 	// the UDP client.
 	Query(q Query) (*Response, error)
+
+	// Close releases any resources a Client may be using.
+	Close() error
```

Comment:
In my experience, the contract on New or Open has always been that if `err` is non-nil then you must not call Close, and if it is nil then you must (should) call Close in the future.

Maybe the real solution here is to have the user create their own `net.Conn` and pass it in. That way they can tweak whatever knobs they want and the burden of connection lifecycle is passed on to them.